packages/pwa-kit-react-sdk/src/utils/opentelemetry.js

```diff
@@ -0,0 +1,337 @@
+/*
+ * Copyright (c) 2025, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {trace, context, SpanStatusCode} from '@opentelemetry/api'
+import {hrTimeToMilliseconds, hrTimeToTimeStamp} from '@opentelemetry/core'
+import logger from './logger-instance'
+
+const DEFAULT_SERVICE_NAME = 'pwa-kit-react-sdk'
+
+export const OTEL_CONFIG = {
+    serviceName: process.env.OTEL_SERVICE_NAME || DEFAULT_SERVICE_NAME,
+    enabled: process.env.OTEL_SDK_ENABLED === 'true',
+    b3TracingEnabled: process.env.OTEL_B3_TRACING_ENABLED === 'true'
+}
+
+export const getServiceName = () => OTEL_CONFIG.serviceName
+
+const SERVICE_NAME = OTEL_CONFIG.serviceName
+
+const logSpanData = (span, event = 'start', res = null) => {
+    const spanContext = span.spanContext()
+    const startTime = span.startTime
+    const endTime = event === 'start' ? startTime : span.endTime
+    const duration = event === 'start' ? 0 : hrTimeToMilliseconds(span.duration)
+
+    // Create the span data object that matches the expected format
+    const spanData = {
+        traceId: spanContext.traceId,
+        parentId: span.parentSpanId,
+        name: span.name,
+        id: spanContext.spanId,
+        kind: span.kind,
+        timestamp: hrTimeToTimeStamp(startTime),
+        duration: duration,
+        attributes: {
+            'service.name': SERVICE_NAME,
+            ...span.attributes,
+            event: event // Add event type to distinguish start/end
+        },
+        status: {code: event === 'start' ? SpanStatusCode.UNSET : SpanStatusCode.OK},
+        events: [],
+        links: [],
+        start_time: startTime,
+        end_time: endTime,
+        forwardTrace: OTEL_CONFIG.b3TracingEnabled
+    }
+
+    // Inject B3 headers into response if available
+    if (res && process.env.DISABLE_B3_TRACING !== 'true' && event === 'start') {
+        res.setHeader('x-b3-traceid', spanContext.traceId)
+        res.setHeader('x-b3-spanid', spanContext.spanId)
+        res.setHeader('x-b3-sampled', '1')
+
+        if (span.parentSpanId) {
+            res.setHeader('x-b3-parentspanid', span.parentSpanId)
+        }
+    }
+
+    // Only log if this is an end event or if it's a start event for a new span
+    if (event === 'end' || !Object.prototype.hasOwnProperty.call(span.attributes, 'event')) {
+        logger.info('OpenTelemetry span data', {
+            namespace: 'opentelemetry.logSpanData',
+            additionalProperties: spanData
+        })
+    }
+}
+
+/**
+ * Creates a new span with the given name and options
+ * @param {string} name - The name of the span
+ * @param {Object} options - Span options
+ * @returns {Span} The created span
+ */
+export const createSpan = (name, options = {}) => {
+    try {
+        const tracer = trace.getTracer(SERVICE_NAME)
```

Comment:
For example, we would replace it here:
```suggestion
        const tracer = trace.getTracer(getServiceName())
```